elvesAndroid/src/main/java/com/idealista/android/elves/view/Adapter.java

```diff
@@ -10,18 +10,34 @@
 import com.idealista.android.elves.view.widget.OnClicked;
 import java.util.Iterator;
 
-public class Adapter<VItemModel> extends RecyclerView.Adapter<Adapter.Holder> {
+public class Adapter<VItemModel> extends RecyclerView.Adapter<Holder> {
 
+    private static final int TYPE_HEADER = 0;
+    private static final int TYPE_ITEM = 1;
+    private static final int TYPE_FOOTER = 2;
     private final CustomViewCreator<VItemModel> customViewCreator;
     private ListableModel<VItemModel> list;
     private final OnClicked<VItemModel> onClickListener;
+    private CustomView headerCustomView;
+    private CustomView footerCustomView;
+    private boolean hasToShowHeader;
```

Comment:
Maybe, but if we do it that way, we force the user to set the custom view every time he wants to show/hide the header/footer. This way, if the user sets it using the builder, and then he hide the header, he wouldn't have to set the same view again before calling show method.

I think that those control variables make our life a bit easier :smile: 
What do you think?